Updated known issue for inclusive_scan_by_key compiler bug (#513)
ROCmSoftwarePlatform/rocThrust: CHANGELOG.md

* Fixed the HIP backend not passing `TestCopyIfNonTrivial` from the upstream (thrust) test suite.
* Fixed tests failing when compiled with `-D_GLIBCXX_ASSERTIONS=ON`.

### Known Issues

> * `thrust::inclusive_scan_by_key` on occasion will produce the incorrect output when -O2 or -O3 optimization is utilized.  

Reviewer: ```suggestion
*  `thrust::inclusive_scan_by_key` might produce incorrect results when it's used with -O2 or -O3 optimization.  
```

If you can specify how it's incorrect, that would be good. That way the user knows what to look out for.

Author: I don't think its possible to specify how it's incorrect, just that the output will be different from expected. It appears to be different every time.
